packages/core/src/asset/ResourceManager.ts

```diff
@@ -182,17 +182,48 @@ export class ResourceManager {
   /**
    * @internal
    */
-  _onSubAssetSuccess<T>(assetURL: string, value: T): void {
-    this._subAssetPromiseCallbacks[assetURL]?.resolve(value);
-    delete this._subAssetPromiseCallbacks[assetURL];
+  _onSubAssetSuccess<T>(assetBaseURL: string, assetSubPath: string, value: T): void {
+    const subPromise = this._subAssetPromiseCallbacks[assetBaseURL]?.[assetSubPath];
+    if (subPromise) {
+      subPromise.resolve(value);
+    } else {
+      (this._subAssetPromiseCallbacks[assetBaseURL] ||= {})[assetSubPath] = {
+        resolve: value
+      };
+    }
   }
 
   /**
    * @internal
    */
-  _onSubAssetFail(assetURL: string, value: (reason: any) => void): void {
-    this._subAssetPromiseCallbacks[assetURL]?.reject(value);
-    delete this._subAssetPromiseCallbacks[assetURL];
+  _onSubAssetFail(assetBaseURL: string, assetSubPath: string, value: (reason: any) => void): void {
+    const subPromise = this._subAssetPromiseCallbacks[assetBaseURL]?.[assetSubPath];
+    if (subPromise) {
+      subPromise.reject(value);
+    } else {
+      (this._subAssetPromiseCallbacks[assetBaseURL] ||= {})[assetSubPath] = {
+        reject: value
+      };
+    }
+  }
+
+  /**
+   * @internal
+   */
+  _createSubAssetPromiseCallback<T>(assetBaseURL: string, assetSubPath: string): AssetPromise<T> {
+    return new AssetPromise<T>((resolve, reject) => {
+      const subPromise = this._subAssetPromiseCallbacks[assetBaseURL]?.[assetSubPath];
+      if (subPromise?.resolve) {
+        resolve(subPromise.resolve);
+      } else if (subPromise?.reject) {
+        reject(subPromise.reject);
+      } else {
+        (this._subAssetPromiseCallbacks[assetBaseURL] ||= {})[assetSubPath] = {
+          resolve,
+          reject
+        };
+      }
+    });
```

Comment:
`_createSubAssetPromiseCallback`、`_onSubAssetSuccess`、`_onSubAssetFail`：The design of these three functions is confusing